### PR TITLE
[WEB-3364] Storybook migration: remove Rails, setup GH Pages deployment

### PR DESCRIPTION
**Pull request by jamiehenson**
Jira Ticket Link / Motivation
----

This PR is 4 of 6 when it comes to the Storybook migration. With #302, we now have a functioning Storybook integration for our Ably UI assets. This PR is all about gutting out all the old Rails assets that are no longer used, and providing a cleaner way of deploying the new assets - in this case, using Github Actions to deploy the static Storybook site to Github Pages.

Jira link: [https://ably.atlassian.net/browse/WEB-3364](https://ably.atlassian.net/browse/WEB-3364)

After this is a TSX rewrite of the Ably UI JSX components, and an investigation into swapping out Webpack as the build tool in favour of Vite.

Summary of changes
----

- Deployment path to GitHub Pages added to GitHub actions, which triggers on release
- Linting/formatting checks moved to GitHub Actions
- CircleCi config stubbed to run with a noop, to satisfy checks until it can be disabled for this repo
- Rails preview folder removed
- All references to Rails/Ruby processes, and Ruby assets removed

How do you manually test this?
----

Pull it down, get the latest packages, make sure Storybook still runs. Upon release, we'll see whether the deployment works correctly (note: I tested this exact deployment on branch push, and it worked - deploying successfully to https://ably.github.io/ably-ui.

Reviewer Tasks (optional)
----

Merge/Deploy Checklist
----

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

Frontend Checklist
----

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

### CODE REVIEW

File: public/mockServiceWorker.js [hunk 1]
```diff
@@ -8,121 +8,121 @@
  * - Please do NOT serve this file on production.
```
Comment (jamiehenson) [inline public/mockServiceWorker.js:8]: All changes here due to linting requirements

File: pull_request_template.md [hunk 2]
```diff
@@ -1,28 +1,21 @@
-Jira Ticket Link / Motivation
```
Comment (jamiehenson) [inline pull_request_template.md:1]: All changes here due to linting requirements

File: src/pages/Chips.mdx [hunk 3]
```diff
@@ -5,58 +5,132 @@ import Icon from "../core/Icon/component.jsx";
```
Comment (jamiehenson) [inline src/pages/Chips.mdx:5]: All changes here due to linting requirements

**Review by kennethkalmer — APPROVED** [review]
Looks great @jamiehenson! well done on being well on your way to having a total net-negative contribution 🤣🎖️

**Pull request merged**